lib/Sema/CodeSynthesis.cpp

```diff
@@ -2090,8 +2122,21 @@ static bool allPropertyWrapperValueSettersAreAccessible(VarDecl *var) {
   return true;
 }
 
-static void finishPropertyWrapperImplInfo(VarDecl *var) {
+static void finishPropertyWrapperImplInfo(VarDecl *var,
+                                          StorageImplInfo &info) {
   auto parentSF = var->getDeclContext()->getParentSourceFile();
+  if (!parentSF)
+    return;
+
+  // Properties with wrappers must not declare a getter or setter.
+  if (!info.hasStorage() && parentSF->Kind != SourceFileKind::Interface) {
+    auto &ctx = parentSF->getASTContext();
+    for (auto attr : var->getAttrs().getAttributes<CustomAttr>())
+      ctx.Diags.diagnose(attr->getLocation(), diag::property_wrapper_computed);
```

Comment:
Can you use the request to look for a property-wrapper attribute instead of assuming that `CustomAttr` necessarily means a property-wrapper attribute?  This diagnostic wouldn't apply to an arbitrary custom attribute.

Reply:
Sure.